### PR TITLE
Add policy specific functionality

### PR DESCRIPTION
**Pull request by charithag**

### CODE REVIEW

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/policy.js [hunk 1]
```diff
@@ -38,33 +38,47 @@ policyModule = function () {
...
+    privateMethods.publish = function (queName, policyName, deviceType, policyDefinition) {
+        log.warn("Queue : " + queName);
+
+        var configurationService = utility.getConfigurationService();
+        var mqttEndPointDeviceConfig = configurationService.getControlQueue(constants.MQTT_QUEUE_CONFIG_NAME);
+        var mqttBrokerURL = mqttEndPointDeviceConfig.getServerURL();
+        var mqttBrokerPort = mqttEndPointDeviceConfig.getPort();
+        var mqttQueueEndpoint = mqttBrokerURL + ":" + mqttBrokerPort;
+
+        var mqttsenderClass = Packages.org.wso2.carbon.device.mgt.iot.mqtt.PolicyPush;
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/policy.js:73]: Please use camelcase notation to specify var names.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                                                                                 data[i].createdTimeStamp
+                                                                             ]);
+                        }
+                    }
+                }
+
+            };
+            invokerUtil.get(serviceURL,
+                            successCallback, function (message) {
+                        console.log(message);
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:94]: Is it necessary to log here?

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                    }
+                    viewModel.policies = policyListToView;
+                    var content = template(viewModel);
+                    $("#policy-list-container").html(content);
+                    $('#policy-table').datatables_extended();
+                }
+            };
+            invokerUtil.get(serviceURL,
+                            successCallback, function (message) {
+                        console.log(message);
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:149]: Is it necessary to log here?

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            };
+
+            var successCallbackPolicy = function (data) {
+                data = JSON.parse(data);
+                $('#policy-spinner').addClass('hidden');
+                if (data != null && data.active == true) {
+                    activePolicy = data;
+                    invokerUtil.get(serviceURLCompliance,
+                                    successCallbackCompliance, function (message) {
+                                console.log(message);
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:191]: Is it necessary to log here?

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                            });
+                }
+            };
+
+            invokerUtil.get(serviceURLPolicy,
+                            successCallbackPolicy, function (message) {
+                        console.log(message);
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:198]: Is it necessary to log here?

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+        $.template("policy-view", policySrc, function (template) {
+            var serviceURLPolicy = "/devicemgt_admin/policies/" + deviceType + "/" + deviceId + "/active-policy"
+            var serviceURLCompliance = "/devicemgt_admin/policies/" + deviceType + "/" + deviceId;
+
+            var successCallbackCompliance = function (data) {
+                var viewModel = {};
+                viewModel.policy = activePolicy;
+                viewModel.deviceType = deviceType;
+                if (data != null && data.complianceFeatures != null && data.complianceFeatures != undefined && data.complianceFeatures.length > 0) {
```
Comment (harshanL) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:170]: I think you can simplify here using ,
if (data && data.complianceFeatures  && data.complianceFeatures.length > 0)

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js (hunk 5 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:198]: This is error call back. So it is better to log the error.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/policy.js (hunk 1 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/modules/policy.js:73]: Ack, Fixed in commit https://github.com/wso2/carbon-device-mgt/pull/303/commits/051493db6a3fdb2e75cc7d1f48cb3ed407622312

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js (hunk 2 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:94]: This is error call back. So it is better to log the error.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js (hunk 3 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:149]: This is error call back. So it is better to log the error.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js (hunk 4 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:191]: This is error call back. So it is better to log the error.

File: features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js (hunk 6 above)
Comment (charithag) [inline features/cdmf-base-ui/org.wso2.carbon.devicemgt.ui.feature/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.view/public/js/device-view.js:170]: Ack, Fixed in commit https://github.com/wso2/carbon-device-mgt/pull/303/commits/051493db6a3fdb2e75cc7d1f48cb3ed407622312

**Pull request merged**